crates/environ/src/fact/trampoline.rs

```diff
@@ -256,8 +402,282 @@ struct Memory<'a> {
     offset: u32,
 }
 
-impl Compiler<'_, '_> {
-    fn compile_adapter(
+impl<'a, 'b> Compiler<'a, 'b> {
+    fn new(module: &'b mut Module<'a>, result: FunctionId, nlocals: u32) -> Self {
+        Self {
+            types: module.types,
+            module,
+            result,
+            code: Vec::new(),
+            nlocals,
+            free_locals: HashMap::new(),
+            traps: Vec::new(),
+            fuel: INITIAL_FUEL,
+            emit_resource_call: false,
+        }
+    }
+
+    fn compile_async_to_async_adapter(
+        mut self,
+        adapter: &AdapterData,
+        start: FunctionId,
+        return_: FunctionId,
+        param_count: i32,
+    ) {
+        let enter = self.module.import_async_enter_call();
+        let exit = self
+            .module
+            .import_async_exit_call(adapter.lift.options.callback, None);
+
```

Comment:
Mind adding a comment here as a small recap of what these intrinsics are doing? For example it looks a little odd to immediately call one then the other but I presume the host is doing something as part of enter/exit which may continue to execute wasm code?

Reply:
I've added doc comments to `fact.rs` to describe the purpose of these intrinsics.  Let me know if they need more work.